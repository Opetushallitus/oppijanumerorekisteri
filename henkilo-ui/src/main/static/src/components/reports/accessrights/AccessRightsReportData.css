.reportScroll {
    overflow-x: scroll;
    margin-top: 1em;
    padding-bottom: 1em;
    transform: rotateX(180deg);
}
.reportWrapper {
    width: 120em;
    transform: rotateX(180deg);
}
.right {
    text-align: right;
}
