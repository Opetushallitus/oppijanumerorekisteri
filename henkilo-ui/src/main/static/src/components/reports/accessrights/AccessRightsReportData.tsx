import React from 'react';
import { Link } from 'react-router';
import type { AccessRightsReportRow } from '../../../reducers/report.reducer';
import type { TableHeading } from '../../../types/react-table.types';
import moment from 'moment';
import Table from '../../common/table/Table';
import PropertySingleton from '../../../globals/PropertySingleton';
import './AccessRightsReportData.css';

type Props = {
    translate: (string) => string;
    report: AccessRightsReportRow[];
};

export const formatDate: (string) => string = (value) => moment(value).format(PropertySingleton.state.PVM_FORMAATTI);

export const formatDateTime: (string) => string = (value) =>
    moment(value).format(PropertySingleton.state.PVM_DATE_TIME_FORMAATTI);

export const columns: TableHeading[] = [
    {
        key: 'personName',
        label: 'HENKILO_NIMI',
        maxWidth: 300,
    },
    {
        key: 'personOid',
        label: 'HENKILO_OPPIJANUMERO',
        maxWidth: 250,
        Cell: (cellProps) => <Link to={`/virkailija/${cellProps.value}`}>{cellProps.value}</Link>,
    },
    {
        key: 'organisationName',
        label: 'HENKILO_KAYTTOOIKEUS_ORGANISAATIO_TEHTAVA',
        maxWidth: 300,
    },
    {
        key: 'organisationOid',
        label: 'HENKILO_OID',
        maxWidth: 250,
    },
    {
        key: 'accessRightName',
        label: 'HENKILO_KAYTTOOIKEUS_KAYTTOOIKEUS',
    },
    {
        key: 'startDate',
        label: 'HENKILO_KAYTTOOIKEUS_ALKUPVM',
        maxWidth: 120,
        Cell: (cellProps) => <div className="right">{formatDate(cellProps.value)}</div>,
    },
    {
        key: 'endDate',
        label: 'HENKILO_KAYTTOOIKEUS_LOPPUPVM',
        maxWidth: 135,
        Cell: (cellProps) => <div className="right">{formatDate(cellProps.value)}</div>,
    },
    {
        key: 'modified',
        label: 'OPPIJOIDEN_TUONTI_LUONTIAIKA',
        maxWidth: 165,
        Cell: (cellProps) => <div className="right">{formatDateTime(cellProps.value)}</div>,
    },
    {
        key: 'modifiedBy',
        label: 'VIRKAILIJAN_TIEDOT_OTSIKKO',
        maxWidth: 250,
        Cell: (cellProps) => <Link to={`/virkailija/${cellProps.value}`}>{cellProps.value}</Link>,
    },
];

export const AccessRightsReport: React.FC<Props> = ({ report, translate }) => (
    <div className="reportScroll">
        <div className="henkilohakuTableWrapper reportWrapper">
            <Table
                headings={columns.map((column) => ({ ...column, label: translate(column.label) }))}
                data={report || []}
                noDataText={translate('HENKILO_KAYTTOOIKEUS_VOIMASSAOLEVAT_TYHJA')}
                resizable
                striped
                highlight
            />
        </div>
    </div>
);

const AccessRightsReportWrapper: React.FC<Props> = (props) => (props.report ? <AccessRightsReport {...props} /> : null);

export default AccessRightsReportWrapper;
