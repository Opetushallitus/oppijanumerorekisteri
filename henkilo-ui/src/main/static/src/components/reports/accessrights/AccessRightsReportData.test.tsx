import { shallow } from 'enzyme';
import AccessRightsReportData, { AccessRightsReport, formatDate } from './AccessRightsReportData';

// mock translation functionality
const translate = (x: string) => x;

describe('AccessRightsReportData', () => {
    test('Should not render anything when report is undefined', () => {
        expect(shallow(<AccessRightsReportData report={undefined} translate={translate} />)).toMatchInlineSnapshot(
            '""'
        );
    });

    test('Should render (empty) report without crashing', () => {
        expect(shallow(<AccessRightsReportData report={[]} translate={translate} />)).toMatchSnapshot();
    });
});

describe('AccessRightsReport', () => {
    test('Should render (empty) report without crashing', () => {
        expect(shallow(<AccessRightsReport report={[]} translate={translate} />)).toMatchSnapshot();
    });
});

describe('formatDate', () => {
    test('format dates correctly', () => {
        expect(formatDate('2022-01-05T16:20:00')).toEqual('5.1.2022');
    });
});
