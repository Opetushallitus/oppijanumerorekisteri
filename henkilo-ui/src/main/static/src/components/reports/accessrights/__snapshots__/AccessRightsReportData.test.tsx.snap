// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessRightsReport Should render (empty) report without crashing 1`] = `
<div
  className="reportScroll"
>
  <div
    className="henkilohakuTableWrapper reportWrapper"
  >
    <Table
      data={Array []}
      fetchMoreSettings={Object {}}
      headings={
        Array [
          Object {
            "key": "personName",
            "label": undefined,
            "maxWidth": 300,
          },
          Object {
            "Cell": [Function],
            "key": "personOid",
            "label": undefined,
            "maxWidth": 250,
          },
          Object {
            "key": "organisationName",
            "label": undefined,
            "maxWidth": 300,
          },
          Object {
            "key": "organisationOid",
            "label": undefined,
            "maxWidth": 250,
          },
          Object {
            "key": "accessRightName",
            "label": undefined,
          },
          Object {
            "Cell": [Function],
            "key": "startDate",
            "label": undefined,
            "maxWidth": 120,
          },
          Object {
            "Cell": [Function],
            "key": "endDate",
            "label": undefined,
            "maxWidth": 135,
          },
        ]
      }
      highlight={true}
      resizable={true}
      striped={true}
    />
  </div>
</div>
`;

exports[`AccessRightsReportData Should render (empty) report without crashing 1`] = `
<AccessRightsReport
  report={Array []}
  translate={[MockFunction]}
/>
`;
